notebooks/haystack-qa.ipynb

```diff
@@ -0,0 +1,494 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #24.            details="minimum" ## Choose from <code>minimum</code>, <code>medium</code>, and <code>all</code>
</pre><p>Looks great!! Thanks Pep.</p><p>I was observing some of the contexts. I do have some observations and suggestions in this context. </p><ol><li>"All" is the choice means whole document ?</li><li>I also observed that some of the question got included in the context along with the answers. what if we separate the questions from the answers in the corpus text. and train our model with the text only containing the answers and not question. lmkwyt? I will also try to apply it in my QA model testing work.   </li></ol>
<br/>

 _Reply via <a href='https://app.reviewnb.com/redhat-et/foundation-models-for-documentation/pull/12/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Reply:
<p><code>all</code> <a href="https://docs.haystack.deepset.ai/reference/utils-api#print_answers" target="_blank">means</a>: show all the details about each answer, e.g. score, offset, ... vs <code>minimum</code> which only shows the answer and context. I thought "minimum" would be enough here, but happy to change it to show all the details.</p><p> </p><p>About context that includes questions: yes, that comes from some of the source documents that do include questions. Most notably, the FAQ in the ROSA worshop: https://www.rosaworkshop.io/rosa/14-faq/</p><p> </p><p>If we try to pre-process the content to separate Questions from Answers, I'm not sure how successful that would be in general because most of the answers need their respective questions in order to make sense (extreme case: there are answers that are just "Yes"). But for that document in particular (the FAQ) it might make sense to use it for the "squad-like" test... only that I have not verified if all the answers there can be obtained from other documents.</p>